pkg/nats/stream/producer_client.go

```diff
@@ -0,0 +1,223 @@
+package stream
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/nats-io/nats.go"
+	"github.com/rs/zerolog/log"
+)
+
+type ProducerClientParams struct {
+	Conn   *nats.Conn
+	Config StreamProducerClientConfig
+}
+
+type ProducerClient struct {
+	Conn *nats.Conn
+	mu   sync.RWMutex // Protects access to activeStreamInfo and activeConnHeartBeatRequestSubjects
```

Comment:
Based on the usage of this lock it appears a regular mutex (`sync.Mutex`) would be sufficient. Is there something I am missing that requires this to be a `RWMutex`?